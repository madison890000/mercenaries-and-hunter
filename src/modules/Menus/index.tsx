import * as React from 'react';
import MenuList from '@mui/material/MenuList';
import MenuItem from '@mui/material/MenuItem';
import {useLocation, useNavigate} from "react-router";
import {defineMessages, useIntl} from "react-intl";
import SelectLanguage from "../SelectLanguage";
import LocaleContext from '../../contexts/LocaleContext';

const messages = defineMessages({
    web: {
        id: 'menu.assistant-job',
        defaultMessage: 'Assist Find Job'
    },
    home: {
        id: 'menu.send-list',
        defaultMessage: 'Applied List'
    },
    tools: {
        id: 'menu.tools',
        defaultMessage: 'Tools'
    },
});


const Menus = () => {
    const navigate = useNavigate();
    const location = useLocation();
    const intl = useIntl();
    const MENU_LIST = [
        {name: 'web', hash: '/web', title: intl.formatMessage(messages.web)},
        {name: 'home', hash: '/', title: intl.formatMessage(messages.home)},
        {name: 'tools', hash: '/tools', title: intl.formatMessage(messages.tools)},
    ];

    const {locale, updateLocale, languages} = React.useContext(LocaleContext);
    return (
        <div style={{
            position: 'relative'
        }}>
            <MenuList
                style={{
                    display: 'flex',
                    justifyContent: 'center',
                    flexWrap: 'wrap',
                    color: 'var(--color-gray)',
                    background: 'var(--color-white)'
                }}
            >
                {
                    MENU_LIST.map((e) => (
                        <MenuItem
                            key={e?.hash}
                            onClick={() => {
                                navigate(e?.hash)
                            }}
                            selected={location.pathname === e?.hash}
                        >
                            {e?.title}
                        </MenuItem>
                    ))
                }
            </MenuList>
            <div style={{
                position: 'absolute',
                bottom: 10,
                right: 5,
                zIndex: 1000
            }}>
                <SelectLanguage value={locale} options={languages} onChange={updateLocale}/>
            </div>
        </div>
    );
};
export default Menus;
