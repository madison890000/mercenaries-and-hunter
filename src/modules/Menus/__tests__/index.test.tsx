import React from 'react';
import Menus from '../index';
import renderer from 'react-test-renderer';

test('renders Menus', () => {
    const MenusDom = renderer.create(<Menus />).toJSON();
    expect(MenusDom).toMatchSnapshot();
});
