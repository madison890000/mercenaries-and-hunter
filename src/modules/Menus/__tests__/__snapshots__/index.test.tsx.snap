// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Menus 1`] = `
<ul
  className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
  onKeyDown={[Function]}
  role="menu"
  style={
    Object {
      "display": "flex",
      "flexWrap": "wrap",
    }
  }
  tabIndex={-1}
>
  <li
    className="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="menuitem"
    tabIndex={0}
  >
    <button
      aria-haspopup="true"
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      data-testid="select-language-btn"
      disabled={false}
      id="composition-button"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    />
  </li>
  <li
    aria-disabled={true}
    className="MuiButtonBase-root MuiMenuItem-root Mui-disabled MuiMenuItem-gutters Mui-disabled MuiMenuItem-root Mui-disabled MuiMenuItem-gutters css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="menuitem"
    tabIndex={-1}
  >
    Download PDF
  </li>
  <li
    aria-disabled={true}
    className="MuiButtonBase-root MuiMenuItem-root Mui-disabled MuiMenuItem-gutters Mui-disabled MuiMenuItem-root Mui-disabled MuiMenuItem-gutters css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="menuitem"
    tabIndex={-1}
  >
    Import From LinkedIn
  </li>
  <li
    aria-disabled={true}
    className="MuiButtonBase-root MuiMenuItem-root Mui-disabled MuiMenuItem-gutters Mui-disabled MuiMenuItem-root Mui-disabled MuiMenuItem-gutters css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="menuitem"
    tabIndex={-1}
  >
    Themes
  </li>
</ul>
`;
